cassovary-core/src/main/scala/com/twitter/cassovary/graph/labels/Label.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.twitter.cassovary.graph.labels
+
+import scala.collection.mutable
+import scala.reflect.runtime.universe.TypeTag
+
+/**
+ * A trait to support adding labels to nodes and edges.
+ * A label is of any type, has a name and maps an 'id' (such as node-id or edge-id) to a value.
+ *
+ * @tparam ID the id-type of the node or edges (e.g., Int)
+ * @tparam L  the type of the label value
+ */
+trait Label[ID, L] extends mutable.Map[ID, L] {
+  val name: String
+  val tag: TypeTag[L]
+}
+
+// the simplest of labels backed by a mutable Hashmap
+class MapBasedLabel[ID, L](val name: String)(implicit val tag: TypeTag[L])
+    extends mutable.HashMap[ID, L] with Label[ID, L]
+
+// mix in this trait to make a label that is defined only for some ids
+trait PartialLabel[ID, L] extends Label[ID, L] {
+  def underlying: mutable.Set[ID]
+  private def isThere(id: ID): Boolean = underlying.contains(id)
```

Comment:
PartialLabel sounds similar to PartialFunction maybe isThere should be isDefinedAt then?